# Questions 1:
How to import pandas and check the version?

In [1]:
'''importing pandas'''
import pandas
'''checking version'''
pandas.__version__

'1.0.5'

# Questions 2:
How to create a series from a numpy array?

In [8]:
import numpy as np
import pandas as pd
'''generating array'''
a = np.array([[1,2,3], [4,5,6],[7,8,9]])
'''reshaping array and creating Series'''
b = a.reshape([9,])
c = pd.Series(b)
'''printing the Series'''
print(c)

0    1
1    2
2    3
3    4
4    5
5    6
6    7
7    8
8    9
dtype: int32


# Questions 3:
How to convert the index of a series into a column of a dataframe?

In [29]:
a = pd.DataFrame(c, columns=['val'])
c = a.reset_index(level = 0)
c

index  val
0      0    1
1      1    2
2      2    3
3      3    4
4      4    5
5      5    6
6      6    7
7      7    8
8      8    9

# Questions 4:
Write the code to list all the datasets available in seaborn library.
Load the 'mpg' dataset
Note: mpg dataset will be read from seaborn module in the manner sir has already shown(provided in the
materials folder)

In [45]:
import seaborn as sns
import os
os.chdir('S://Coding//AIML//seaborn-data-master//Datasets//')
'''to see datasets available in Seaborn library'''
print('datasets in Seaborn are ',os.listdir())
df = sns.load_dataset('mpg')
df.head(5)

datasets in Seaborn are  ['anagrams.csv', 'anscombe.csv', 'attention.csv', 'brain_networks.csv', 'car_crashes.csv', 'diamonds.csv', 'dots.csv', 'exercise.csv', 'flights.csv', 'fmri.csv', 'gammas.csv', 'geyser.csv', 'iris.csv', 'mpg.csv', 'penguins.csv', 'planets.csv', 'tips.csv', 'titanic.csv']


mpg  cylinders  displacement  horsepower  weight  acceleration  \
0  18.0          8         307.0       130.0    3504          12.0   
1  15.0          8         350.0       165.0    3693          11.5   
2  18.0          8         318.0       150.0    3436          11.0   
3  16.0          8         304.0       150.0    3433          12.0   
4  17.0          8         302.0       140.0    3449          10.5   

   model_year origin                       name  
0          70    usa  chevrolet chevelle malibu  
1          70    usa          buick skylark 320  
2          70    usa         plymouth satellite  
3          70    usa              amc rebel sst  
4          70    usa                ford torino

# Questions 5:
Which country origin cars are a part of this dataset?

In [48]:
'''We use the .value_counts() function to see the unique values and their counts'''
print(df['origin'].value_counts())

usa       249
japan      79
europe     70
Name: origin, dtype: int64

#### based on the above info, cars originating in usa, japan and europe are mentioned in the dataset

# Questions 6:
Extract the part of the dataframe which contains cars belonging to 'usa'

In [53]:
'''applying filter for 'origin' column'''
df_usa = df[df['origin'] == 'usa']
print(df_usa.shape)
df_usa.head(10)

(249, 9)


mpg  cylinders  displacement  horsepower  weight  acceleration  \
0  18.0          8         307.0       130.0    3504          12.0   
1  15.0          8         350.0       165.0    3693          11.5   
2  18.0          8         318.0       150.0    3436          11.0   
3  16.0          8         304.0       150.0    3433          12.0   
4  17.0          8         302.0       140.0    3449          10.5   
5  15.0          8         429.0       198.0    4341          10.0   
6  14.0          8         454.0       220.0    4354           9.0   
7  14.0          8         440.0       215.0    4312           8.5   
8  14.0          8         455.0       225.0    4425          10.0   
9  15.0          8         390.0       190.0    3850           8.5   

   model_year origin                       name  
0          70    usa  chevrolet chevelle malibu  
1          70    usa          buick skylark 320  
2          70    usa         plymouth satellite  
3          70    usa              amc rebel sst  
4          70    usa                ford torino  
5          70    usa           ford galaxie 500  
6          70    usa           chevrolet impala  
7          70    usa          plymouth fury iii  
8          70    usa           pontiac catalina  
9          70    usa         amc ambassador dpl

#### The above data-set 'df_usa' consists of only those records where the origin is usa.
#### We see that there are 249 records of cars originating in usa.